ui-v2/src/api/task-runs/task-runs.test.ts

```diff
@@ -0,0 +1,72 @@
+import { createFakeTaskRun } from "@/mocks";
+import { QueryClient, useSuspenseQuery } from "@tanstack/react-query";
+import { renderHook, waitFor } from "@testing-library/react";
+import { buildApiUrl, createWrapper, server } from "@tests/utils";
+import { http, HttpResponse } from "msw";
+import { describe, expect, it } from "vitest";
+import { TaskRun, TaskRunsFilter, buildListTaskRunsQuery } from ".";
+
+describe("task runs api", () => {
+	const mockFetchTaskRunsAPI = (taskRuns: Array<TaskRun>) => {
+		server.use(
+			http.post(buildApiUrl("/task_runs/filter"), () => {
+				return HttpResponse.json(taskRuns);
+			}),
+		);
+	};
+
+	describe("taskRunsQueryParams", () => {
+		it("fetches paginated task runs with default parameters", async () => {
+			const taskRun = createFakeTaskRun();
+			mockFetchTaskRunsAPI([taskRun]);
+
+			const queryClient = new QueryClient();
+			const { result } = renderHook(
+				() => useSuspenseQuery(buildListTaskRunsQuery()),
+				{ wrapper: createWrapper({ queryClient }) },
+			);
+
+			await waitFor(() => {
+				expect(result.current.data).toEqual([taskRun]);
+			});
+		});
+
+		it("fetches paginated task runs with custom search parameters", async () => {
+			const taskRun = createFakeTaskRun();
+			mockFetchTaskRunsAPI([taskRun]);
+
+			const filter: TaskRunsFilter = {
+				offset: 0,
+				limit: 10,
+				sort: "ID_DESC" as const,
+				task_runs: {
+					operator: "and_" as const,
+					name: { like_: "test-task-run" },
+				},
+			};
+
+			const queryClient = new QueryClient();
+			const { result } = renderHook(
+				() => useSuspenseQuery(buildListTaskRunsQuery(filter)),
+				{ wrapper: createWrapper({ queryClient }) },
+			);
+
+			await waitFor(() => expect(result.current.isSuccess).toBe(true));
+			expect(result.current.data).toEqual([taskRun]);
+		});
```

Comment:
Is this test valuable? We're not mocking any of the filter behavior right? The filter could specify `name: { like_: 'not-this-task-run' }` and it would still return it wouldn't it?